modules/heat_conduction/src/kernels/HeatConductionTimeDerivative.C

```diff
@@ -17,7 +17,7 @@ InputParameters validParams<HeatConductionTimeDerivative>()
   // over current volume by setting the use_displaced_mesh flag.
   params.set<bool>("use_displaced_mesh") = true;
 
-  params.addParam<MaterialPropertyName>("specific_heat", "specific_heat",
+  params.addParam<MaterialPropertyName>("specific_heat_name", "specific_heat",
```

Comment:
We do not have to shove _all_ the documentation into the parameter names themselves, we do have doc strings (and the actual parameter type is available to the UIs that help people write input files).